src/treetensornetworks/projttns/projttnsum.jl

```diff
@@ -1,17 +1,25 @@
 """
 ProjTTNSum
 """
-struct ProjTTNSum{V}
-  pm::Vector{ProjTTN{V}}
-  function ProjTTNSum(pm::Vector{ProjTTN{V}}) where {V}
-    return new{V}(pm)
+struct ProjTTNSum{T<:AbstractProjTTN}
+  pm::Vector{T} where {T}
+  function ProjTTNSum(pm::Vector{T}) where {T}
+    return new{T}(pm)
```

Comment:
```suggestion
  function ProjTTNSum(pm::Vector)
    return new{eltype(pm)}(pm)
```
I generally prefer that style, using functions to access type parameters instead of `where` when possible.

Reply:
Also maybe this should be constrained to accept `AbstractProjTTN`, i.e.:
```julia
  function ProjTTNSum(pm::Vector{<:AbstractProjTTN})
    # ...
  end
```